okta/request_executor.py

```diff
@@ -259,13 +260,39 @@ async def fire_request_helper(self, request, attempts, request_start_time):
             date_time = headers.get("Date", "")
             if date_time:
                 date_time = convert_date_time_to_seconds(date_time)
+
+            # Get X-Rate-Limit-Reset header
             retry_limit_reset_headers = list(map(float, headers.getall(
                 "X-Rate-Limit-Reset", [])))
             # header might be in lowercase, so check this too
             retry_limit_reset_headers.extend(list(map(float, headers.getall(
                 "x-rate-limit-reset", []))))
             retry_limit_reset = min(retry_limit_reset_headers) if len(
                 retry_limit_reset_headers) > 0 else None
+
+            # Get X-Rate-Limit-Limit Header
+            retry_limit_limit_headers = list(map(float, headers.getall(
+                "X-Rate-Limit-Limit", [])))
+            # header might be in lowercase, so check this too
+            retry_limit_limit_headers.extend(list(map(float, headers.getall(
+                "x-rate-limit-limit", []))))
+            retry_limit_limit = min(retry_limit_limit_headers) if len(
+                retry_limit_limit_headers) > 0 else None
+
+            # Get X-Rate-Limit-Remaining Header
+            retry_limit_remaining_headers = list(map(float, headers.getall(
+                "X-Rate-Limit-Remaining", [])))
+            # header might be in lowercase, so check this too
+            retry_limit_remaining_headers.extend(list(map(float, headers.getall(
+                "x-rate-limit-remaining", []))))
+            retry_limit_remaining = min(retry_limit_remaining_headers) if len(
+                retry_limit_remaining_headers) > 0 else None
+
+            # both X-Rate-Limit-Limit and X-Rate-Limit-Remaining being 0 indicates concurrent rate limit error
+            if retry_limit_limit is not None and retry_limit_remaining is not None:
+                if retry_limit_limit == 0 and retry_limit_remaining == 0:
+                    logger.warning('Concurrent limit rate exceeded')
```

Comment:
Are there tests that already cover this logic?  Are there tests that can be added if not?

Reply:
Good callout @bryanapellanes-okta! I'll update a test and add another one for the changes